EKF/covariance.cpp

```diff
@@ -260,18 +260,25 @@ void Ekf::predictCovariance()
 
 	// assign IMU noise variances
 	// inputs to the system are 3 delta angles and 3 delta velocities
+	const float gyro_noise = math::constrain(_params.gyro_noise, 0.0f, 1.0f);
 	float daxVar, dayVar, dazVar;
-	float dvxVar, dvyVar, dvzVar;
-	float gyro_noise = math::constrain(_params.gyro_noise, 0.0f, 1.0f);
 	daxVar = dayVar = dazVar = sq(dt * gyro_noise);
+
 	float accel_noise = math::constrain(_params.accel_noise, 0.0f, 1.0f);
+	if (_imu_sample_delayed.delta_vel_clip_count > 0 && _imu_sample_delayed.delta_vel_samples > 0) {
+		// total clip count across every axis
+		const float clipping_fraction = _imu_sample_delayed.delta_vel_clip_count / (_imu_sample_delayed.delta_vel_samples * 3);
+
+		accel_noise += 2.0f * clipping_fraction;
```

Comment:
Let's make this a tuning parameter so we can use replay to iterate on a suitable value.

Reply:
I've added _params->accel_noise_clipping_factor, which is EKF2_ACC_CLPFACT PX4 side.